core/src/test/java/org/apache/iceberg/TestCreateTransaction.java

```diff
@@ -268,9 +268,13 @@ public void testCreateDetectsUncommittedChangeOnCommit() throws IOException {
 
     txn.updateProperties().set("test-property", "test-value"); // not committed
 
-    AssertHelpers.assertThrows("Should reject commit when last operation has not committed",
-        IllegalStateException.class, "Cannot commit transaction: last operation has not committed",
-        txn::commitTransaction);
+    txn.commitTransaction();
+
+    // nothing to commit.
+    Assert.assertNull("Starting a create transaction should not commit metadata",
+        TestTables.readMetadata("uncommitted_change"));
+    Assert.assertNull("Should have no metadata version",
+        TestTables.metadataVersion("uncommitted_change"));
```

Comment:
I think this test case is still valid.

The problem is that this was previously assuming that the table metadata would change when each of the operations in a transaction is committed. That's not necessarily the case.

Instead of getting rid of the check that each operation committed, we just need to track that it committed differently. I think that means we should replace `lastBase` in `BaseTransaction` with a boolean `hasComitted` that gets set to false each time a new operation is created inside of `checkLastOperationCommitted`.

Reply:
Hi, I have replaced the lastBase with a boolean value to check if last commit has committed. Now It will not change the behaviors of commitTransaction.